airgun/session.py

```diff
@@ -253,318 +272,318 @@ def take_screenshot(self):
     def reporttemplate(self):
         """Instance of Report Template entity."""
-        return ReportTemplateEntity(self.browser)
+        return self._open(ReportTemplateEntity)
 
     @cached_property
     def redhatrepository(self):
         """Instance of Red Hat Repository entity."""
-        return RedHatRepositoryEntity(self.browser)
+        return self._open(RedHatRepositoryEntity)
 
     @cached_property
     def repository(self):
         """Instance of Repository entity."""
-        return RepositoryEntity(self.browser)
+        return self._open(RepositoryEntity)
 
     @cached_property
     def role(self):
         """Instance of Role entity."""
-        return RoleEntity(self.browser)
+        return self._open(RoleEntity)
 
     @cached_property
     def settings(self):
         """Instance of Settings entity."""
-        return SettingsEntity(self.browser)
+        return self._open(SettingsEntity)
 
     @cached_property
     def sc_parameter(self):
         """Instance of Smart Class Parameter entity."""
-        return SmartClassParameterEntity(self.browser)
+        return self._open(SmartClassParameterEntity)
 
     @cached_property
     def smartvariable(self):
         """Instance of Smart Variable entity."""
-        return SmartVariableEntity(self.browser)
+        return self._open(SmartVariableEntity)
 
     @cached_property
     def subnet(self):
         """Instance of Subnet entity."""
-        return SubnetEntity(self.browser)
+        return self._open(SubnetEntity)
 
     @cached_property
     def subscription(self):
         """Instance of Subscription entity."""
-        return SubscriptionEntity(self.browser)
+        return self._open(SubscriptionEntity)
 
     @cached_property
     def syncplan(self):
         """Instance of Sync Plan entity."""
-        return SyncPlanEntity(self.browser)
+        return self._open(SyncPlanEntity)
 
     @cached_property
     def sync_status(self):
         """Instance of Sync Status entity"""
-        return SyncStatusEntity(self.browser)
+        return self._open(SyncStatusEntity)
 
     @cached_property
     def task(self):
         """Instance of Task entity."""
-        return TaskEntity(self.browser)
+        return self._open(TaskEntity)
 
     @cached_property
     def trend(self):
         """Instance of Trend entity."""
-        return TrendEntity(self.browser)
+        return self._open(TrendEntity)
 
     @cached_property
     def user(self):
         """Instance of User entity."""
-        return UserEntity(self.browser)
+        return self._open(UserEntity)
 
     @cached_property
     def usergroup(self):
         """Instance of User Group entity."""
-        return UserGroupEntity(self.browser)
+        return self._open(UserGroupEntity)
 
     @cached_property
     def virtwho_configure(self):
         """Instance of Virtwho Configure entity."""
-        return VirtwhoConfigureEntity(self.browser)
+        return self._open(VirtwhoConfigureEntity)
```

Comment:
This comment is not a blocker, the PR LGTM. 

I only have one comment/suggestion about how it is possible to make it DRY, all that `@cached_property` are similiar and too much code repetition.

```python
@lru_cache  # lru_cache can replace @cached_property
def get_open_entity(self, name):
    # This dict can actually be automatically built from `airgun.entities` members.
    entities = {
        'virtwho_configure': VirtwhoConfigureEntity,
       'usergroup': UserGroupEntity,
      ...
    }
    return self._open(entities[name])

def __getattr__(self, name):
    """Called when an attribute lookup has not found"""
    try:
        return self.get_open_entity(name)
    except KeyError:
        return super().__getattr__(name)
```

That is just an idea to be tested/evaluated and maybe implemented in another PR  because with this it is possible to replace all the repetition, 248 lines becomes 72 lines. (or less)

Any time a `session.foo` is called, the `foo` is going to be looked up in the entity list before it raises `AttributeError` if not found.

Reply:
@rochacbruno looks like to be a good idea not sure here about consequences, unrelated to this PR. can you please create an issue in Airgun with mentioning details.

Reply:
https://github.com/SatelliteQE/airgun/issues/426